File: tools/claude/config/skills/scan-git-for-tils/op/secrets.py
"""1Password secret retrieval utilities."""

from __future__ import annotations

import subprocess

OP_NOTION_TOKEN_PATH = "op://Scripts/Notion/api-access-token"


def get_op_secret(path: str) -> str:
    """Fetch a secret from 1Password."""
    result = subprocess.run(["op", "read", path], capture_output=True, text=True)

    if result.returncode != 0:
        raise RuntimeError(f"Failed to retrieve secret from 1Password: {result.stderr.strip()}")

    return result.stdout.strip()

File: tools/claude/config/skills/scanning-git-for-tils/op/secrets.ts
/**
 * 1Password secret retrieval - BUN VERSION
 */

export const OP_NOTION_TOKEN = "op://Scripts/Notion/api-access-token";

export async function getOpSecret(path: string): Promise<string> {
  const proc = Bun.spawn(["op", "read", path], {
    stdout: "pipe",
    stderr: "pipe",
  });

  const exitCode = await proc.exited;
  if (exitCode !== 0) return "";

  const output = await new Response(proc.stdout).text();
  return output.trim();
}
